src/index/flat/flat.cc

```diff
@@ -236,7 +237,7 @@ class FlatIndexNode : public IndexNode {
     bool
     HasRawData(const std::string& metric_type) const override {
         if constexpr (std::is_same<T, faiss::IndexFlat>::value) {
-            return !IsMetricType(metric_type, metric::COSINE);
+            return true;
         }
         if constexpr (std::is_same<T, faiss::IndexBinaryFlat>::value) {
             return true;
```

Comment:
curious: why doesn't this if have an else branch? this doesn't seem legit c++ code to me.. how come it compiles before?

Reply:
There are only 2 possibilities to run into here, one is faiss::IndexFlat, the other is faiss::IndexBinaryFlat, so no else